aea/mail/base.py

```diff
@@ -178,38 +208,262 @@ def __str__(self):
             .format(to=self.to, sender=self.sender, protocol_id=self.protocol_id, message=self.message)
 
 
+class Multiplexer:
+    """This class can handle multiple connections at once."""
+
+    def __init__(self, connections: List['Connection'], in_queue: queue.Queue,
+                 default_connection_index: int = 0, loop: Optional[AbstractEventLoop] = None):
```

Comment:
`default_connection_index` is used for envelopes which don't specify any routing context.